src/App.tsx

```diff
@@ -15,48 +15,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type SortOptions = {
+  sortType: SortType,
+  reverse: boolean,
+};
+
+export function getPreparedGoods(
+  goods: string[],
+  { sortType, reverse }: SortOptions,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return SortType.NONE;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortField] = useState(SortType.NONE);
+  const [reverse, setSortReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer, { sortType, reverse },
+  );
+
+  const reset = () => {
+    setSortField(SortType.NONE);
+    setSortReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortType !== SortType.ALPHABET && 'is-light'}`}
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally. Fix all cases